### PR TITLE
sql: avoid invalidating hash shard check constraints in `IMPORT INTO`

### LINKED ISSUE #89012: sql/importer: hash shard check constraints are invalidated with IMPORT INTO
**Describe the problem**

Check constraints get marked as `NOT VALID` when running `IMPORT INTO`. This is what it is, but in general could be okay if we documented it (https://github.com/cockroachdb/docs/issues/15244). 

This is particularly painful when it comes to hash sharded indexes. Those things have associated check constraints, but we do not show them in the create statements. The good news about those constraints is that they are valid by construction. That means we can detect such constraints in the importer and choose not to mark them as unvalidated here:

https://github.com/cockroachdb/cockroach/blob/ef4656fc14155afd31a486b38f688c0dee08ca0b/pkg/sql/importer/import_job.go#L1006-L1009

**To Reproduce**
Import into a hash sharded table and notice that the shard constraints are invalid by looking at `pg_constraint`. 

**Additional context**
Potentially we should find a way to show the fact that the constraints are not valid to the user -- perhaps by choosing to show the constraint in `CREATE TABLE` statements if the constraint is `NOT VALID`. That would be an easy change, though we have to ask what we ought to do if the user gives us such a statement -- do we reject it? Do we accept it but actually go and validate the constraint when we make the new table?

Epic CRDB-8353

Jira issue: CRDB-20076

### PR DESCRIPTION
**Pull request by ZhouXing19**
Previously, with IMPORT INTO, when we updates the status of imported tables from
OFFLINE to PUBLIC, all constraints are marked unvalidated. This will change the
implicit check constraints accompanied by hash-shared indexes as well, but
the user won't notice that it is `NOT VALID` now from `SHOW CREATE TABLE`.

This commit is to make sure these special check constraints won't be marked as
unvalidated when executing IMPORT INTO.

fixes https://github.com/cockroachdb/cockroach/issues/89012

Release note (bug fix):  exclude check constraints of hash-shared index from
being unvalidated when executing IMPORT INTO

### CODE REVIEW

File: pkg/sql/importer/import_job.go [hunk 1]
```diff
@@ -1005,7 +1005,11 @@ func (r *importResumer) publishTables(
 
 				// Set CHECK constraints to unvalidated before publishing the table imported into.
 				for _, c := range newTableDesc.AllActiveAndInactiveChecks() {
-					c.Validity = descpb.ConstraintValidity_Unvalidated
+					// We only "unvalidate" constraints that are not hash-sharded column
+					// check constraints.
+					if !c.Hidden {
```
Comment (ajwerner) [inline pkg/sql/importer/import_job.go:1010]: should we rename this field? The comment on it doesn't make a ton of sense to me.

File: pkg/sql/delegate/show_table.go [hunk 2]
```diff
@@ -91,6 +91,11 @@ SELECT
 				ELSE
 					e'\n-- Warning: Partitioned table with no zone configurations.'
         END
+        ,
+        /* TODO(janexing): How to avoid printing the contraints that's already in the create stmt */
+        CASE WHEN (SELECT string_agg(details, E'\n') FROM [SHOW CONSTRAINTS FROM %[7]s] WHERE validated='f' AND constraint_type != 'PRIMARY KEY') IS NOT NULL THEN 
+            concat(E'\n', (SELECT string_agg(details, E'\n') FROM [SHOW CONSTRAINTS FROM %[7]s] WHERE validated='f' AND constraint_type != 'PRIMARY KEY'), ' NOT VALID') 
+        END
```
Comment (ajwerner) [inline pkg/sql/delegate/show_table.go:98]: I don't think this is the place to make the change. You should change the logic which generates the `CREATE TABLE` statement rather that doing anything up here in SQL. Also, you can (and probably should) have two separate PRs for these two changes.

**Review by ajwerner — APPROVED** [review]
LGTM

**Comment by ZhouXing19** [comment]
TFTR!
bors r=ajwerner

**Pull request merged**